lib/SILOptimizer/IPO/GlobalOpt.cpp

```diff
@@ -1455,10 +1455,13 @@ bool SILGlobalOpt::run() {
     GlobIdx = 0;
     for (auto &BB : F) {
       bool IsCold = ColdBlocks.isCold(&BB);
-      auto Iter = BB.begin();
-      while (Iter != BB.end()) {
-        SILInstruction *I = &*Iter;
-        Iter++;
+
+      llvm::SmallVector<SILInstruction *, 4> Worklist;
```

Comment:
I wish SmallVector in LLVM could accept ranges. Then you could just do:

```
llvm::SmallVector<SILInstruction *, 4> Worklist(reverse(BB));
```

(Of course this is outside of the scope of this radar.

Reply:
Can you hoist the Worklist out of the inner loop and just assert that it is empty at the top?